data/data/gcp/main.tf

```diff
@@ -16,13 +17,15 @@ module "bootstrap" {
 
   bootstrap_enabled = var.gcp_bootstrap_enabled
 
-  image        = google_compute_image.cluster.self_link
-  machine_type = var.gcp_bootstrap_instance_type
-  cluster_id   = var.cluster_id
-  ignition     = var.ignition_bootstrap
-  network      = module.network.network
-  subnet       = module.network.master_subnet
-  zone         = var.gcp_master_availability_zones[0]
+  image            = google_compute_image.cluster.self_link
+  machine_type     = var.gcp_bootstrap_instance_type
+  cluster_id       = var.cluster_id
+  ignition         = var.ignition_bootstrap
+  network          = module.network.network
+  network_cidr     = var.machine_cidr
```

Comment:
this should be from the the network module.

Reply:
The machine_cidr? Aside from the input machine_cidr, the network module is not aware of this value.. as networks don't have cidr. It's subnets each have their own cidr. If we are using machine_cidr anyhow... it should be passed from the source as I have done here.

Reply:
So the details about the network are controlled by the network module, the CIDR range for the network _ even though technically is a list of CIDR blocks for subnets_ is still something that the network module should provide the information.

today we create and provide the cidr ranges for subnets from main.tf when actually these should be completely under the control of the network module because for new subnets those values are generated by us, but for pre-existing subnets those values are fetched from API.

So keeping this in network module means others can just depend on that to provide it a list, values could be anything and don't have to be tied into the root modules input.

Reply:
This was a pre-existing case... can we spawn another card to cover it?

Reply:
sure, we can fix it as follow-up.